datadog/fwprovider/data_source_datadog_app_builder_app.go

```diff
@@ -0,0 +1,197 @@
+package fwprovider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var _ datasource.DataSource = &appBuilderAppDataSource{}
+
+type appBuilderAppDataSource struct {
+	Api  *datadogV2.AppBuilderApi
+	Auth context.Context
+}
+
+func NewDatadogAppBuilderAppDataSource() datasource.DataSource {
+	return &appBuilderAppDataSource{}
+}
+
+func (d *appBuilderAppDataSource) Configure(_ context.Context, request datasource.ConfigureRequest, response *datasource.ConfigureResponse) {
+	providerData := request.ProviderData.(*FrameworkProvider)
+	d.Api = providerData.DatadogApiInstances.GetAppBuilderApiV2()
+	// Used to identify requests made from Terraform
+	d.Api.Client.Cfg.AddDefaultHeader("X-Datadog-App-Builder-Source", "terraform")
```

Comment:
Not sure if this is required. You could use `User-Agent` to identify requests being made by Terraform clients. I see a similar approach to yours is already used in `workflow_automation` resource/datasource, but that's the only occurrence.

Is `User-Agent` not enough for your use case?

Reply:
Ah yes I was copying the approach done in `workflow_automation`. As discussed offline, you are right that `User-Agent` should be enough to identify Terraform requests, and it's already included by default.

https://github.com/DataDog/terraform-provider-datadog/blob/master/datadog/fwprovider/framework_provider.go#L388-L394

https://github.com/DataDog/terraform-provider-datadog/blob/master/datadog/internal/utils/utils.go#L230-L236